### PR TITLE
Forgot to assert bool returns on token transfers

### PR DESCRIPTION
**Pull request by fubuloubu**

### CODE REVIEW

**Comment by spinoch** [comment]
Could you add a test for this?

File: contracts/Vault.vy [hunk 1]
```diff
@@ -791,7 +791,7 @@ def withdraw(_shares: uint256 = MAX_UINT256, _recipient: address = msg.sender) -
     log Transfer(msg.sender, ZERO_ADDRESS, shares)
```
Comment (spinoch) [inline contracts/Vault.vy:791]: Wouldn't you want to assert this also?

Comment (fubuloubu) [reply contracts/Vault.vy:791]: logs appear in the transaction receipt. broadly, we should probably test for event logs occurring correctly, but haven't gotten to that yet.

**Comment by fubuloubu** [comment]
> Could you add a test for this?

There's a series of weird token patterns that this code will revert on, so it's important that the deployer understands the token that they are using for their vault

**Review by eosioAdminmanager — APPROVED** [review]
@0xa60e9d48477bf0fd37fa2857aa904c09e6d26444

**Review by eosioAdminmanager — APPROVED** [review]

File: contracts/Vault.vy (hunk 1 above)
Comment (spinoch) [reply contracts/Vault.vy:791]: How about changing this to
```suggestion
    assert self.transfer(ZERO_ADDRESS, shares)
```
?

Comment (spinoch) [reply contracts/Vault.vy:791]: And if I remember correctly there's another place where `log Transfer` is used and this could also be done

Comment (fubuloubu) [reply contracts/Vault.vy:791]: Ah, I think I see the confusion. This function is taking place of a mint function, it doesn't actually transfer any shares but creates them. This is a weird aspect of using ERC20.

Comment (spinoch) [reply contracts/Vault.vy:791]: Ah, right. That somewhat makes sense

**Review by lbertenasco — APPROVED** [review]

**Pull request merged**